src/appleseed.shaders/src/alshaders/alSurface.osl

```diff
@@ -293,12 +293,15 @@ shader alSurface
     int do_glossy2 = 1;
     int do_glossy = 1;
 
-    if (raytype("transparency"))
+    if (raytype("transparency") || raytype("shadow"))
     {
-        out_result = (color(1) - in_opacity) * transparent();
-        return;
+        if (min(in_opacity) < 1.0)
```

Comment:
@est77 why the usage of min()? OSL inequality test will test component by component, so a != 1 would suffice, since there is no opacity value above 1.0, any != 1.0 value would be < 1.0 necessarily.